scripts/dev-e2e.sh

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+
+set -euo pipefail
+
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m'
+
+STATUSFILE='armadactl-query.txt'
+AOREPO='https://github.com/armadaproject/armada-operator.git'
+ARMADA_CTL_VERSION='0.18.3'
+JOB_DETAILS=1
+
+usage() {
+  cat > /dev/stderr  <<USAGE
+Usage: ${0} -s
+  -s => get job status details from Armada
+
+USAGE
+  exit 1
+}
+
+while getopts "s" opt; do
+    case "${opt}" in
+        s )
+          JOB_DETAILS=1 ;;
+        * )
+          echo "opt is $opt"
+          usage ;;
+    esac
+done
+
+scripts="$(cd "$(dirname "$0")"; pwd)"
+source "$scripts"/init.sh
+
+AOHOME="$scripts/../../armada-operator"
+
+log() {
+  echo -e "${GREEN}$1${NC}"
+}
+
+err() {
+  echo -e "${RED}$1${NC}" >&2
+}
+
+
+fetch-armadactl() {
+  os=$(uname -s)
+  arch=$(uname -m)
+  dl_url='https://github.com/armadaproject/armada/releases/download'
+
+  if [ "$os" = "Darwin" ]; then
+    dl_os='darwin'
+  elif [ "$os" = "Linux" ]; then
+    dl_os='linux'
+  else
+    err "fetch-armadactl(): sorry, operating system $os not supported; exiting now"
+    exit 1
+  fi
+
+  if [ "$arch" = 'arm64' ]; then
+    dl_arch='arm64'
+  elif [ "$arch" = 'x86_64' ]; then
+    dl_arch='amd64'
+  else
+    err "fetch-armadactl(): sorry, architecture $arch not supported; exiting now"
+    exit 1
+  fi
+
+  dl_file=armadactl_${ARMADA_CTL_VERSION}_${dl_os}_${dl_arch}.tar.gz
+
+  curl --silent --location "${dl_url}/v${ARMADA_CTL_VERSION}/$dl_file" > "$scripts/$dl_file"
+  (cd "$scripts" && tar xzf "$dl_file" armadactl)
```

Comment:
I think this way is even better to do:
```suggestion
  tar xzf "$scripts/$dl_file" -C "$scripts" armadactl)
```

Reply:
Oh yes, much better - nice to lose the subshell and the chdir - fixed in commit 704accd.